cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs

```diff
@@ -0,0 +1,222 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Testnet.Test.Cli.Queries
+  ( hprop_cli_queries
+  ) where
+
+import           Cardano.Api
+import qualified Cardano.Api as Api
+
+import           Cardano.CLI.Types.Output (QueryTipLocalStateOutput)
+import           Cardano.Testnet
+import qualified Cardano.Testnet.Test.Cli.Conway.DRepRetirement as DRep
+
+import           Prelude
+
+import           Control.Monad (forM_, void)
+import           Control.Monad.IO.Class
+import qualified Data.Aeson as Aeson
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as Text
+import qualified Data.Vector as Vector
+import           GHC.IORef (newIORef)
+import           GHC.Stack (HasCallStack, withFrozenCallStack)
+import           System.FilePath ((</>))
+
+import qualified Testnet.Process.Cli as H
+import qualified Testnet.Process.Run as H
+import qualified Testnet.Property.Utils as H
+import           Testnet.Property.Utils (queryUtxos)
+import           Testnet.Runtime
+
+import           Hedgehog
+import qualified Hedgehog as H
+import           Hedgehog.Extras (Integration)
+import qualified Hedgehog.Extras as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Hedgehog.Extras.Test.Golden as H
+
+-- | Test CLI queries
+-- Execute me with:
+-- @cabal test cardano-testnet-test --test-options '-p "/CliQueries/"'@
+-- If you want to recreate golden files, run the comment with
+-- RECREATE_GOLDEN_FILES=1 as its prefix
+hprop_cli_queries :: Property
+hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' -> do
+  conf@Conf { tempAbsPath=tempAbsPath@(TmpAbsolutePath work) }
+    <- mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  utxoFileCounter <- liftIO $ newIORef 1
+
+  let sbe = ShelleyBasedEraConway
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoSlotLength = 0.1
+        , cardanoNodeEra = cEra
+        }
+
+  TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets
+    , configurationFile
+    }
+    <- cardanoTestnetDefault fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+
+  let queryAnyUtxo :: Text.Text -> Integration TxIn
```

Comment:
This is obsolete now.

Reply:
Changed to use `findLargestUtxoForPaymentKey` :+1: